### PR TITLE
handle a custom public directory option

### PR DESCRIPTION
**Pull request by tomjn**
Allow users to change `public_html` to something else, defaultiing to public_html

### CODE REVIEW

**Review by Mte90 — DISMISSED** [review]
As I can see use as default public_html as default so is fine for me

**Comment by tomjn** [comment]
We need the readme updating, and a test provsion. Ideally to demonstrate `htdocs` as an example

**Comment by msaggiorato** [comment]
I'm totally interested in this, but shouldn't we also allow to avoid public_html altogether? and just provision directly to the site folder?

**Comment by tomjn** [comment]
> I'm totally interested in this, but shouldn't we also allow to avoid public_html altogether? and just provision directly to the site folder?

I don't think anything prevents though, if it did it would probably be the trailng slash in nginx, just set the value to `""`

File: provision/vvv-nginx-default.conf [hunk 1]
```diff
@@ -2,7 +2,7 @@ server {
     listen       80;
     listen       443 ssl http2;
     server_name  {vvv_hosts};
-    root         {vvv_path_to_site}/public_html;
+    root         {vvv_path_to_site}/{vvv_public_dir};
```
Comment (msaggiorato) [inline provision/vvv-nginx-default.conf]: Wouldn't this become a problem?

Comment (msaggiorato) [reply provision/vvv-nginx-default.conf]: In the context of an empty public_dir.

Comment (tomjn) [reply provision/vvv-nginx-default.conf]: you mean with no value? No idea, would need testing

**Comment by tomjn** [comment]
I tested this with:

```yml
  public:
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    branch: "add/custom-public_html"
    hosts:
      - public.test
    custom:
      public_dir: htdocs
```

**Comment by msaggiorato** [comment]
I'll give this a go

**Comment by tomjn** [comment]
I'm particularly interested in empty values, and multiply nested folders

**Comment by msaggiorato** [comment]
Added 3 changes here which makes the following combinations work:

```yml
  public-default:
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    branch: "add/custom-public_html"
    hosts:
      - public-default.test
  
  public-empty:
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    branch: "add/custom-public_html"
    hosts:
      - public-empty.test
    custom:
      public_dir: ''
  
  public-nested:
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    branch: "add/custom-public_html"
    hosts:
      - public-nested.test
    custom:
      public_dir: 'htdocs/nested'

  public-nested-deep:
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    branch: "add/custom-public_html"
    hosts:
      - public-nested-deep.test
    custom:
      public_dir: 'htdocs/nested/deep'

  public-not-nested:
    repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
    branch: "add/custom-public_html"
    hosts:
      - public-not-nested.test
    custom:
      public_dir: 'htdocs'
```

Basically removed all the path forcing and generate a proper wp-cli.yml that can work from the site directory.

PD: In order for the empty case to work, we need https://github.com/Varying-Vagrant-Vagrants/VVV/pull/2201 merged

**Comment by Mte90** [comment]
Cool, so if we are generating wp-cli.yml we can work on this https://github.com/Varying-Vagrant-Vagrants/VVV/issues/1968 after this pr

**Comment by tomjn** [comment]
These `wp-cli.yml` are per site though, https://github.com/Varying-Vagrant-Vagrants/VVV/issues/1968 requires a VM-wide file in the root so it'd need to be generated via the vagrant file

File: provision/vvv-init.sh [hunk 2]
```diff
@@ -187,11 +187,19 @@ update_wp() {
   fi
 }
 
+setup_cli() {
+  rm -f wp-cli.yml
+  if [ ! -z "${PUBLIC_DIR}" ]; then
+    echo "path: ${PUBLIC_DIR}" > wp-cli.yml
```
Comment (tomjn) [inline provision/vvv-init.sh]: Does this need to be wrapped in quotes?

**Comment by tomjn** [comment]
> PD: In order for the empty case to work, we need Varying-Vagrant-Vagrants/VVV#2201 merged

Does this change work without that being merged? If it doesn't then we need to force all users to upgrade to `develop` which is a non-starter

**Comment by msaggiorato** [comment]
> Does this change work without that being merged? If it doesn't then we need to force all users to upgrade to develop which is a non-starter

It does work, but it won't support the empty string case. Without the PR merged on develop, it'll default to public_html.

File: provision/vvv-init.sh (hunk 2 above)
Comment (msaggiorato) [reply provision/vvv-init.sh]: Added quotes

File: provision/vvv-init.sh [hunk 3]
```diff
@@ -183,30 +192,36 @@ update_wp() {
   fi
 }
 
-setup_database
-setup_nginx_folders
-
-cd "${VVV_PATH_TO_SITE}/public_html"
+setup_cli() {
```
Comment (tomjn) [inline provision/vvv-init.sh:195]: we should be explicit about the path this file is in, in future we'll want to consider using a template system similar to the nginx config so users can add other things in here, but not in this PR

**Comment by Mte90** [comment]
It was missing a mention in the readme

**Pull request merged**